### ISSUE TITLE
[Engine] Implement JAX-based force computation experiment

### ISSUE THREAD

**Issue by stevemikedan** [stevemikedan/CosmoFrame#73]
Rebuilt gravitational force calculation using jax.numpy and tested accuracy.
